bsp_diff/caas/device/intel/civ/host/vm-manager/0046-Multi-Android-VM-support.patch

```diff
@@ -0,0 +1,382 @@
++		read input
++		if [ $input = "yes" ]; then
++	                rm -rf $CIV_DIR/$VM_NAME
++	        else
++	                exit
++	        fi
++        fi
++        echo "Creating Dir: $CIV_WORK_DIR/$VM_NAME"
++        mkdir $CIV_WORK_DIR/$VM_NAME
+ }
+ 
+-function create_aaf_dir() {
+-        mkdir -p $CIV_WORK_DIR/scripts/aaf
+-}
++function copy_files_for_vm() {
++    echo "Copying file: $CIV_WORK_DIR/$VM_NAME"
++    mkdir -p $CIV_WORK_DIR/$VM_NAME/scripts/aaf
++
++    req_files=("$CIV_WORK_DIR/OVMF.fd"
++                   "$CIV_WORK_DIR/scripts/rpmb_dev")
++    for file in ${req_files[@]}; do
++        if [ ! -f $file ]; then
++            echo "Error: $file file is missing"
++            exit -1
++        fi
++    done
+ 
++    cp $CIV_WORK_DIR/OVMF.fd $CIV_WORK_DIR/$VM_NAME/
++    cp $CIV_WORK_DIR/scripts/rpmb_dev $CIV_WORK_DIR/$VM_NAME/scripts/
++}
+ 
+ #-------------    main processes    -------------
+ 
+@@ -623,10 +659,11 @@ ubu_install_qemu_gvt
+ ubu_build_ovmf_gvt
+ ubu_enable_host_gvt
+ ubu_update_fw
+-
+ install_vm_manager
+-create_aaf_dir
+-setup_civ_ini
++
++create_vm_dir
++setup_civ_ini $VSOCK_ID $ADB_PORT $VM_NAME
++copy_files_for_vm
+ 
+ prepare_required_scripts
+ ubu_install_swtpm
+diff --git a/scripts/setup_multi_civ_vm.sh b/scripts/setup_multi_civ_vm.sh
+new file mode 100755
+index 0000000..5359d8d
+--- /dev/null
++++ b/scripts/setup_multi_civ_vm.sh
+@@ -0,0 +1,141 @@
++#!/bin/bash
++
++# Copyright (c) 2023 Intel Corporation.
++# All rights reserved.
++#
++# SPDX-License-Identifier: Apache-2.0
++
++set -eE
++
++#---------      Global variable     -------------------
++CIV_DIR=$(pwd)
++USER_HOME=$(getent passwd $SUDO_USER | cut -d: -f6)
++VM_NAME=Android-CIV
++VSOCK_ID=3
++NUMBER_OF_VM=1
++ADB_PORT=5555
++INITIAL_VM_NUM=1
++SETUP_HOST_DONE=false
++
++function copy_files_for_vm_flashing() {
++    mkdir -p $CIV_DIR/$3/scripts/aaf
++
++    req_files=("$CIV_DIR/OVMF.fd"
++                   "$CIV_DIR/scripts/rpmb_dev")
++    for file in ${req_files[@]}; do
++        if [ ! -f $file ]; then
++            echo "Error: $file file is missing"
++            exit -1
++        fi
++    done
++
++    cp $CIV_DIR/OVMF.fd $CIV_DIR/$3/
++    cp $CIV_DIR/scripts/rpmb_dev $CIV_DIR/$3/scripts/
++}
++
++function create_vm_dirs() {
++    mkdir $CIV_DIR/$3
++    $CIV_DIR/scripts/setup_civ_ini.sh -v $1 -p $2 -n $3
++}
++
++
++function create_vms() {
++    create_vm_dirs $1 $2 $3
++    copy_files_for_vm_flashing $1 $2 $3
++}
++
++function setup_files() {
++    echo "Creating setup for VMs"
++    if [ -d $USER_HOME/.intel/.civ ]
++    then
++        SETUP_HOST_DONE=true
++    fi
++
++    new_dir=$CIV_DIR/$VM_NAME
++    if [ -n "$(ls -d ${new_dir}? 2> /dev/null )" ]; then
++	dir_arr=$(ls -d ${new_dir}?)
++	directories=( $dir_arr )
++	last_vm=${directories[-1]}
++	final_vm_name="${last_vm##*/}"
++	last_character=${final_vm_name: -1}
++	last_character=$((last_character+1))
++	ADB_PORT=$((5555+((last_character-1)*2)))
```

Comment:
ADB_PORT=5555 is defined at the start of the file.
To avoid multiple definitions....

Please replace:
ADB_PORT=$((5555+((last_character-1)*2)))

With:
ADB_PORT=$((ADB_PORT+((last_character-1)*2)))